pgmpy/factors/Factor.py

```diff
@@ -76,7 +76,11 @@ def __init__(self, variables, cardinality, values):
         >>> phi
         <Factor representing phi(x1:2, x2:2, x3:2) at 0x7f8188fcaa90>
         """
+
         values = np.array(values)
+        
+        if values.dtype != int and values.dtype != float:
```

Comment:
I think it is okay.
If we give some other type both conditions will be true since it is checking not equal to.
Please correct me if I am making a mistake.